spec/controllers/comments_controller_spec.rb

```diff
@@ -0,0 +1,247 @@
+require 'spec_helper'
+
+describe CommentsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+
+  let(:comment) { create(:comment) }
+  let(:unreviewed_comment) { create(:unreviewed_comment) }
+
+  before(:each) do
+    request.env["HTTP_REFERER"] = "/where_i_came_from"
+  end
+
+  describe "GET #add_comment_reply" do
+    context "when comment is unreviewed" do
+      it "redirects logged out user to login path with an error" do
+        get :add_comment_reply, comment_id: unreviewed_comment.id
+        it_redirects_to_with_error(login_path, "Sorry, you cannot reply to an unapproved comment.")
+      end
+
+      it "redirects logged in user to root path with an error" do
+        fake_login
+        get :add_comment_reply, comment_id: unreviewed_comment.id
+        it_redirects_to_with_error(root_path, "Sorry, you cannot reply to an unapproved comment.")
+      end
+    end
+
+    context "when comment is not unreviewed" do
+      it "redirects to the comment on the commentable without an error" do
+        get :add_comment_reply, comment_id: comment.id
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(work_path(comment.ultimate_parent, show_comments: true, anchor: "comment_#{comment.id}"))
+      end
+
+      it "redirects to the comment on the commentable with the reply form open and without an error" do
+        get :add_comment_reply, comment_id: comment.id, id: comment.id
+        expect(flash[:error]).to be_nil
+        expect(response).to redirect_to(work_path(comment.ultimate_parent, add_comment_reply_id: comment.id, show_comments: true, anchor: "comment_#{comment.id}"))
+      end
+    end
+  end
+
+  describe "GET #unreviewed" do
```

Comment:
We don't get that we would throw a 500 for an unfound record ?

Failures:

  1) CommentsController GET #unreviewed On an invalid id it errors
     Failure/Error: @commentable = Comment.find(params[:comment_id])

     ActiveRecord::RecordNotFound:
       Couldn't find Comment with id=999999
     # /home/vagrant/.rvm/gems/ruby-2.2.5/gems/activerecord-3.2.22.3/lib/active_record/relation/finder_methods.rb:344:in `find_one'
     # /home/vagrant/.rvm/gems/ruby-2.2.5/gems/activerecord-3.2.22.3/lib/active_record/relation/finder_methods.rb:315:in `find_with_ids'
     # /home/vagrant/.rvm/gems/ruby-2.2.5/gems/activerecord-3.2.22.3/lib/active_record/relation/finder_methods.rb:107:in `find'
     # /home/vagrant/.rvm/gems/ruby-2.2.5/gems/activerecord-3.2.22.3/lib/active_record/querying.rb:5:in `find'
     # ./app/controllers/comments_controller.rb:132:in `load_commentable'

The fix might be:

https://github.com/otwcode/otwarchive/pull/2716/files but I am not feeling well enough to be sure.